label_sleuth/app.py

```diff
@@ -740,11 +737,27 @@ def import_labels(workspace_id):
 
     :param workspace_id:
     """
+    required_columns = ['text', 'category_name', 'label']
     csv_data = StringIO(request.files['file'].stream.read().decode("utf-8"))
     df = pd.read_csv(csv_data).rename(columns=lambda x: x.strip())
 
-    return jsonify(curr_app.orchestrator_api.import_category_labels(workspace_id, df))
+    # check for required columns and send specific error messaging
+    missing_columns = []
+    for col in required_columns:
+        try: 
+            df[col]
+        except KeyError as e:
+            missing_columns.append(e.__str__())
 
+    if len(missing_columns) > 0:
+        return jsonify({ 'type': 'missing_required_columns', 
+                        'title': f"Missing the following required columns: {','.join(missing_columns)}" }), 400
+
+    # try/except around api functionality to catch all other errors
+    try:
+        return jsonify(curr_app.orchestrator_api.import_category_labels(workspace_id, df))
+    except:
+        return jsonify({ 'type': 'invalid_upload', 'title': "Invalid csv file"}), 400
```

Comment:
Excelent! We really needed this this validation. Thanks. In the future we could add this validation to the `add_documents` endpoint as well.